src/Http/ServerRequest.php

```diff
@@ -499,7 +499,7 @@ public function is($type, ...$args): bool
 
         $type = strtolower($type);
         if (!isset(static::$_detectors[$type])) {
-            return false;
+            throw new CakeException("No detector set for type `{$type}`");
```

Comment:
There should be an `@throws` for this. And should this maybe use `InvalidArgumentException` (the core uses it in various other places where argument x isn't present in y)?

Reply:
I think when the valid set of values can be changed at runtime then it should be a `RuntimeException`. But I don't mind using `InvalidArgumentException` if that's the consensus.

Reply:
`InvalidArgumentException`makes sense to me.